components/controls/FavoriteBase.tsx

```diff
@@ -25,75 +25,26 @@ import { ButtonLink } from './buttons';
 import Tooltip from './Tooltip';
 
 export interface Props {
-  addFavorite: () => Promise<void>;
   className?: string;
   favorite: boolean;
   qualifier: string;
-  removeFavorite: () => Promise<void>;
+  toggleFavorite: () => void;
 }
 
-interface State {
-  favorite: boolean;
-}
-
-export default class FavoriteBase extends React.PureComponent<Props, State> {
-  mounted = false;
-
-  constructor(props: Props) {
-    super(props);
-    this.state = { favorite: this.props.favorite };
-  }
-
-  componentDidMount() {
-    this.mounted = true;
-  }
-
-  componentWillUnmount() {
-    this.mounted = false;
-  }
-
-  toggleFavorite = () => {
-    if (this.state.favorite) {
-      this.removeFavorite();
-    } else {
-      this.addFavorite();
-    }
-  };
-
-  addFavorite() {
-    this.props.addFavorite().then(
-      () => {
-        if (this.mounted) {
-          this.setState({ favorite: true });
-        }
-      },
-      () => {}
-    );
-  }
-
-  removeFavorite() {
-    this.props.removeFavorite().then(
-      () => {
-        if (this.mounted) {
-          this.setState({ favorite: false });
-        }
-      },
-      () => {}
-    );
-  }
-
+export default class FavoriteBase extends React.PureComponent<Props> {
```

Comment:
Crazy thought: should we rename this to `FavoriteButton` or something? I find `FavoriteBase` makes it look like an abstract class.